src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -9,13 +9,13 @@
 
 namespace duckdb {
 
-PhysicalTableScan::PhysicalTableScan(vector<LogicalType> types, TableFunction function_p,
+PhysicalTableScan::PhysicalTableScan(ArenaAllocator &arena, vector<LogicalType> types, TableFunction function_p,
```

Comment:
Can we pass in the PhysicalPlanGenerator or PhysicalPlan instead? It's a big refactor to get this into all the operators - so I would prefer to pass in something we can extend in the future if we need to.